src/lib/local-api.js

```
@@ -1263,6 +1263,97 @@ function createLocalApiHandler({ queuePath }) {
+            redirect: "manual",
+          });
+          if ([301, 302, 303, 307, 308].includes(proxyRes.status)) {
+            const location = proxyRes.headers.get("location");
+            if (!location) break;
+            let next;
+            try { next = new URL(location, currentUrl); } catch { break; }
+            if (!NATIVE_PROXY_ALLOWED_HOSTS.has(next.hostname) || next.protocol !== "https:") {
+              break; // redirect leaves allowlist — return the redirect response as-is
+            }
+            // 303 always switches to GET; others preserve method.
+            if (proxyRes.status === 303) { method = "GET"; }
```

Biome (2.5.0)

[error] Can't assign method because it's a constant.

(lint/correctness/noConstAssign)